### ISSUE TITLE
Sets are reported as type aliases

### ISSUE THREAD

**Issue by Teloah** [fabriciocolombo/sonar-delphi#12]
This code generates "Do not use type aliases" issue for `TSuits`:

``` delphi
unit SetTest;

interface

type
  TSuit = (stHearts, stSpades, stDiamonds, stClubs);
  TSuits = set of TSuit;

implementation

end.
```

**Comment by fabriciocolombo** [comment]
It's a grammar bug.

The code below produces the same AST.

``` delphi
type
  TMyChar = Char;
  TMySetOfChar = set of Char;
```

``` xml
   <TkNewType class="" column="2" line="13" method="" package="">
      TkNewType
      <TMyChar class="" column="2" line="13" method="" package="">
         TMyChar
         <Char class="" column="12" line="13" method="" package="">Char</Char>
      </TMyChar>
   </TkNewType>
   <TkNewType class="" column="2" line="14" method="" package="">
      TkNewType
      <TMySetOfChar class="" column="2" line="14" method="" package="">
         TMySetOfChar
         <Char class="" column="24" line="14" method="" package="">Char</Char>
      </TMySetOfChar>
   </TkNewType>
```

I will take a closer look soon
